open-company/open-company-web
# Queue websocket messages when connection is down.

This change will save all data sent to a web socket if that web socket is closed.  When it returns the saved data will be sent to the server.

To test:
This is meant to resolve the id token mark as read , so we will test that.

- generate a digest url with an id token
- logout of carrot
- open to digest url in your browser
- [ ] does the console have the following?:
```
DEBUG [oc.web.ws.change-client:60] - Sending item/seen for container: 8a40-4df7-8df4 item: 898d-42ab-a06b
raven.inc.js:58 DEBUG [oc.web.ws.change-client:64] - Sending item/read for container: 8a40-4df7-8df4 item: 898d-42ab-a06b
raven.inc.js:58 DEBUG [oc.web.ws.change-client:160] - Push event from server:  [:container/status ()]
raven.inc.js:58 DEBUG [oc.web.ws.change-client:93] - event-handler :container/status
raven.inc.js:58 DEBUG [oc.web.ws.change-client:106] - Status event: ()
raven.inc.js:58 DEBUG [oc.web.ws.change-client:160] - Push event from server:  [:container/status ({:unseen [], :container-id "8a40-4df7-8df4"})]
raven.inc.js:58 DEBUG [oc.web.ws.change-client:93] - event-handler :container/status
raven.inc.js:58 DEBUG [oc.web.ws.change-client:106] - Status event: ({:unseen [], :container-id "8a40-4df7-8df4"})
```

- [ ] in the change service logs is there a mark as read/seen event?
something like 
```
[websocket] item/read for: 898d-42ab-a06b in: 8a40-4df7-8df4 at: 2019-01-15T15:26:54.459Z by: 96b3-4f31-9fe1 / 16f1611e-b28a-417d-8854-1312ecf49d03
```

- check that normal web socket operations work with the change service 
  Different users see changes.  Things are marked as read
- check that normal operations work with notify service
Notifications are received  in client
- check that normal operations work with the interaction service.
Different users see reactions in real time.
